File: scribbleshare-app/scribbleshare-frontend/src/scripts/canvas/canvasUpdate/canvasUpdates/CanvasUpdateDelete.ts
import CanvasUpdate from "../CanvasUpdate.js";
import ByteBuffer from "../../../protocol/ByteBuffer.js";
import {Canvas} from "../../Canvas.js";
import CanvasUpdateType from "../CanvasUpdateType.js";

export default class CanvasUpdateDelete extends CanvasUpdate {
    constructor(byteBuffer: ByteBuffer) {
        super(byteBuffer);
    }

    getType(): CanvasUpdateType {
        return CanvasUpdateType.DELETE;
    }

    draw(canvas: Canvas, dt: number): boolean {
        /*        this.time += dt;
                if (this.dt <= this.time) {
                    canvas.canvasObjectWrappers.delete(this.id);
                    return true;
                }*/
        return false;
    }

    serialize(byteBuffer: ByteBuffer) {
        super.serialize(byteBuffer);
    }

    static create(time: number, id: number) {
        let object: CanvasUpdateDelete = Object.create(this.prototype);
        object.dt = time;
        return object;
    }
}

File: commons/src/main/java/net/stzups/scribbleshare/data/objects/canvas/canvasUpdate/canvasUpdates/CanvasUpdateDelete.java
package net.stzups.scribbleshare.data.objects.canvas.canvasUpdate.canvasUpdates;

import io.netty.buffer.ByteBuf;
import net.stzups.scribbleshare.data.objects.canvas.Canvas;
import net.stzups.scribbleshare.data.objects.canvas.canvasUpdate.CanvasUpdate;
import net.stzups.scribbleshare.data.objects.canvas.canvasUpdate.CanvasUpdateException;
import net.stzups.scribbleshare.data.objects.canvas.canvasUpdate.CanvasUpdateType;
import net.stzups.util.DebugString;

public class CanvasUpdateDelete extends CanvasUpdate {
    public CanvasUpdateDelete(ByteBuf byteBuf) {
        super(byteBuf);
    }

    @Override
    protected CanvasUpdateType getCanvasUpdateType() {
        return CanvasUpdateType.DELETE;
    }

    @Override
    public void update(Canvas canvas, short id) throws CanvasUpdateException {
        if (canvas.getCanvasObjects().remove(id) == null) {
            throw new CanvasUpdateException("CanvasObject does not exist");
        }
    }

    @Override
    public String toString() {
        return DebugString.get(CanvasUpdateDelete.class, super.toString())
                .toString();
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
    }
}
